packages/babel-helpers/src/index.js

```diff
@@ -1,10 +1,208 @@
+import traverse from "babel-traverse";
+import * as t from "babel-types";
 import helpers from "./helpers";
 
-export function get(name) {
-  const fn = helpers[name];
-  if (!fn) throw new ReferenceError(`Unknown helper ${name}`);
+function makePath(path) {
+  const parts = [];
 
-  return fn().expression;
+  for (; path; path = path.parentPath) {
+    parts.push(path.key);
+    if (path.inList) parts.push(path.listKey);
+  }
+
+  // Pop off the final "program" since these paths will be used from the Program NodePath.
+  parts.pop();
+
+  return parts.reverse().join(".");
+}
+
+/**
+ * Given a file AST for a given helper, get a bunch of metadata about it so that Babel can quickly render
+ * the helper is whatever context it is needed in.
+ */
+function getHelperMetadata(file) {
+  const globals = new Set();
+  const localBindingNames = new Set();
+
+  let exportName;
+  let exportPath;
+  const exportBindingAssignments = [];
+
+  traverse(file, {
```

Comment:
Instead of doing a full traversal, we could only iterate over `file.program.body`.

Reply:
I wanted to create the NodePath, but yeah fair. I added a `path.skip()` to ignore anything isn't the root import/exports.